In [5]:
import pandas as pd
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import confusion_matrix, classification_report
from sklearn.model_selection import train_test_split

In [2]:
diabetes_df = pd.read_csv('diabetes_clean.csv')

In [6]:
diabetes_df.head()

pregnancies  glucose  diastolic  triceps  insulin   bmi    dpf  age  \
0            6      148         72       35        0  33.6  0.627   50   
1            1       85         66       29        0  26.6  0.351   31   
2            8      183         64        0        0  23.3  0.672   32   
3            1       89         66       23       94  28.1  0.167   21   
4            0      137         40       35      168  43.1  2.288   33   

   diabetes  
0         1  
1         0  
2         1  
3         0  
4         1

In [15]:
y = diabetes_df["diabetes"].values

In [20]:
X = diabetes_df.drop(["diabetes", "pregnancies", "triceps", "glucose", "diastolic", "insulin", "dpf"], axis=1).values

In [21]:
X

array([[33.6, 50. ],
       [26.6, 31. ],
       [23.3, 32. ],
       ...,
       [26.2, 30. ],
       [30.1, 47. ],
       [30.4, 23. ]])

In [22]:
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state= 42, test_size=.3)

In [23]:
X_train.shape

(537, 2)

In [24]:
knn = KNeighborsClassifier(n_neighbors=6) # instantiate neighbors classifier

In [25]:
knn.fit(X_train, y_train) # fit the model

KNeighborsClassifier(n_neighbors=6)

In [26]:
y_pred = knn.predict(X_test) # predict labels of test data

In [27]:
# Generate the confusion matrix and classification report
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))

[[117  34]
 [ 47  33]]
              precision    recall  f1-score   support

           0       0.71      0.77      0.74       151
           1       0.49      0.41      0.45        80

    accuracy                           0.65       231
   macro avg       0.60      0.59      0.60       231
weighted avg       0.64      0.65      0.64       231



#### Results

The model produced 117 true positives, 33 true negatives, 34 false negatives, and 47 false positives. The classification report shows a better F1-score for the zero class, which represents individuals who do not have diabetes.